owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/drawer/DrawerViewModel.kt

```diff
@@ -62,13 +66,39 @@ class DrawerViewModel(
         return AccountUtils.getCurrentOwnCloudAccount(context)
     }
 
+    fun getUsernameOfAccount(accountName: String): String {
+        return AccountUtils.getUsernameOfAccount(accountName)
+    }
+
     fun setCurrentAccount(context: Context, accountName: String): Boolean {
         return AccountUtils.setCurrentOwnCloudAccount(context, accountName)
     }
 
-    fun deleteUnusedUserDirs(accounts: Array<Account>) {
+    fun removeAccount(context: Context) {
         viewModelScope.launch(coroutinesDispatcherProvider.io) {
-            FileStorageUtils.deleteUnusedUserDirs(accounts)
+            val loggedAccounts = AccountUtils.getAccounts(context)
+            FileStorageUtils.deleteUnusedUserDirs(loggedAccounts)
+
+            val userQuotas = getUserQuotasUseCase.execute(Unit)
+            val loggedAccountsNames = mutableListOf<String>()
+            val totalAccountsNames = mutableListOf<String>()
+            loggedAccounts.forEach { loggedAccount ->
+                loggedAccountsNames.add(loggedAccount.name)
+            }
+            userQuotas.forEach { account ->
+                totalAccountsNames.add(account.accountName)
+            }
+            val removedAccountsNames = mutableListOf<String>()
+            for (accountName in totalAccountsNames) {
+                if (!loggedAccountsNames.contains(accountName)) {
+                    removedAccountsNames.add(accountName)
+                }
+            }
+            removedAccountsNames.forEach { removedAccountName ->
+                removeAccountUseCase.execute(
+                    RemoveAccountUseCase.Params(accountName = removedAccountName)
+                )
+            }
```

Comment:
I think we could do it without that variable, but it's ok! Up to you 🍻 
```suggestion
            for (accountName in totalAccountsNames) {
                if (!loggedAccountsNames.contains(accountName)) {
                    removeAccountUseCase.execute(
                        RemoveAccountUseCase.Params(accountName = accountName)
                    )  
                }
            }
```

Reply:
Yes, it can be done without the variable, but I think keeping it improves understandability since maybe `userQuotas` doesn't make clear enough that we are recovering all the accounts saved in the database, and that the inverse of the intersection between that and `loggedAccounts` are the ones which we actually have to remove